manager/controllers/app/plotter_generator.go

```diff
@@ -201,6 +201,80 @@ func (p *PlotterGenerator) addStep(element *ResolvedEdge, datasetID string, api
 	return steps
 }
 
+// setEmptySinkDataFormat sets empty data format in sink nodes to be the first
+// dataformat supported by the modules sink interface.
+func (p *PlotterGenerator) setEmptySinkDataFormat(solution *Solution) {
```

Comment:
There are several issues with function. First, it goes over all supported sinks of the module while checking in each iteration whether DataFormat is empty. Also, it goes over all nodes while in the calling function we have a specific node that should be fixed.
I would suggest to replace this function with getSupportedFormat(capability) that will return a supported dataFormat of the selected capability, and call it prior to AllocateStorage, i.e. when we have an element that needs storage allocation.

Reply:
We changed the implementation. Thanks